code/scripting/api/objs/hudgauge.cpp

```diff
@@ -190,38 +191,66 @@ ADE_FUNC(isHiRes, l_HudGauge, nullptr, "Returns whether this is a hi-res HUD gau
 	return ade_set_args(L, "b", gauge->isHiRes());
 }
 
-ADE_FUNC(getColor, l_HudGauge, nullptr, "Returns the current color used by this HUD gauge.",
-	"number, number, number, number", "The current color, in red, green, blue, and alpha components from 0 to 255")
+ADE_FUNC(getColor,
+	l_HudGauge,
+	nullptr,
+	"Gets the active 2D drawing color. False to return raw rgb, true to return a color object. Defaults to false.",
+	"number, number, number, number | color",
+	"The current color, in red, green, blue, and alpha components from 0 to 255")
 {
 	HudGauge* gauge;
-	if (!ade_get_args(L, "o", l_HudGauge.Get(&gauge)))
+	bool rc = false;
+	if (!ade_get_args(L, "o|b", l_HudGauge.Get(&gauge), &rc))
 		return ADE_RETURN_NIL;
 
 	auto &c = gauge->getColor();
 
-	return ade_set_args(L, "iiii", static_cast<int>(c.red), static_cast<int>(c.green), static_cast<int>(c.blue), static_cast<int>(c.alpha));
+	if (rc) {
+		return ade_set_args(L, "o", l_Color.Set(c));
+	}else{
```

Comment:
```suggestion
	} else {
```

Just slightly distracting without the spaces